### DISCUSSION TITLE
To inquire about Index.html

### DISCUSSION THREAD

**Discussion by Rucha-Ambaliya** [tjiuce/Dataverse#355] [General]
I'm encountering a dark screen when opening this page, with the website logo displayed in the center and a chatbot at the bottom right corner. However, there is no button available to navigate through the website.
![Screenshot 2024-12-29 170102](https://github.com/user-attachments/assets/94696dfd-8fa6-434a-ab49-10275bc0be75)

Am I supposed to navigate the website via some other file? Please let me know if I'm doing it in the wrong way.

**Comment by tjiuce** [comment]
> I'm encountering a dark screen when opening this page, with the website logo displayed in the center and a chatbot at the bottom right corner. However, there is no button available to navigate through the website.
> ![Screenshot 2024-12-29 170102](https://github.com/user-attachments/assets/94696dfd-8fa6-434a-ab49-10275bc0be75)
> 
> Am I supposed to navigate the website via some other file? Please let me know if I'm doing it in the wrong way.

Hey @Rucha-Ambaliya, ideally the logo remains at the centre for about 3 seconds like a preloader and then the contents of the website are displayed along with the chatbot.

The website is working fine on my device, maybe due to bad network the js file didn't execute on your device. Please check again and let me know if the same issue persists.

**Reply by Rucha-Ambaliya** [reply]
I followed your suggestion, but the issue remains the same. However, when I ran other HTML files directly, they opened perfectly. As mentioned, after loading, the chatbot appears on the screen, but the other content does not.

**Reply by tjiuce** [reply]
@Rucha-Ambaliya I checked this yesterday on multiple devices and couldn't find any problem. I also checked if there is any problem in `script.js` (the file that controls timing of the preloader screen in `index.html`). 

You can fork and try to solve it, feel free to reach out if you find the issue. 

Thanks for your interest in the project though!

**Reply by Rucha-Ambaliya** [reply]
Thank you for your help and time! I deleted the repository, set it up locally again, and it's working now—I believe there might have been an installation mistake on my end.  

Also, I was wondering if it would be useful to include commands for creating a virtual environment in the installation section of the documentation? It might help avoid potential setup issues in the future. Sorry if I missed it, if it’s already there. Let me know your thoughts!  

Thanks again!

**Reply by tjiuce** [reply]
@Rucha-Ambaliya you're absolutely right, we need to update the installation instructions as many changes have been made in the project structure lately. Adding installation commands would be very helpful.

Thanks for the suggestion 👍

**Reply by Rucha-Ambaliya** [reply]
If it’s okay, may I work on that part? I’d be happy to add the installation commands and other instructions as well. Just let me know if there’s anything specific that needs to be included.

**Reply by tjiuce** [reply]
Sure @Rucha-Ambaliya, go ahead! There is a file [installation/requirements](installation/requirements), which includes all the modules that need to be installed before running the [main.py](software/main.py) file. 

So the users basically have to run these commands:


```
cd installation
```

```
pip  install -r requirements.txt
```

You can add the rest of the commands 👍
Thank you!

**Comment by Rucha-Ambaliya** [comment]
About the issue I mentioned earlier of blank screen, I debugged something and if I’m correct, the website isn’t using a local server, right? I believe that’s why the browser was throwing CORS errors and blocking JS files from loading. Upon finding a solution, I realized the 'Go Live' extension is a good fix for this. If I’m right, should I add a reference to this in the documentation? I think new contributors, especially those inclined towards web development, might face a similar issue, and it could be a big hindrance. For instance, when I ran main.py and clicked on the 'Go to Website' button, it opened, and I didn’t realize at the time that it was a deployed website 😅. Pretty naive of me, but I think there will be others who might miss these small details too. What do you think?

Also, now that index.html is running properly on my device, I’m going to actively work on the 'Contribute' button as well.

**Comment by tjiuce** [comment]
@Rucha-Ambaliya you're right, the website isn’t using a local server. Instead of suggesting the 'Go Live' extension to the contributors, we should add a section in the [contributing.md](https://github.com/multiverseweb/Dataverse/blob/main/Documentation/contributing.md) file, informing contributors about the port to open in their browser. 

The website can be accessed by opening 

```
http://localhost:2604/
```
in the browser.

Also, to deal with the CORS errors can be fixed by adding 

```json
{
    "liveServer.settings.CORS": true
}
```
 to the [.vscode/settings.json](.vscode/settings.json) file.
 I have created a new issue #362 for the same, let me know if you want to work on it or discuss this further.

**Comment by Rucha-Ambaliya** [comment]
Sure, I would like to work on it. Thanks for assigning this to me! 
I'll surely discuss with you if I face any problem during solving it.
